src/set_theory/principal.lean

```diff
@@ -103,4 +109,129 @@ theorem unbounded_principal (op : ordinal → ordinal → ordinal) :
   set.unbounded (<) {o | principal op o} :=
 λ o, ⟨_, principal_nfp_blsub₂ op o, (le_nfp_self _ o).not_lt⟩
 
+/-! #### Additive principal ordinals -/
+
+theorem principal_add_one : principal (+) 1 :=
+principal_one_iff.2 $ zero_add 0
+
+theorem principal_add_of_le_one {o : ordinal} (ho : o ≤ 1) : principal (+) o :=
+begin
+  rcases le_one_iff.1 ho with rfl | rfl,
+  { exact principal_zero },
+  { exact principal_add_one }
+end
+
+theorem principal_add_is_limit {o : ordinal} (ho₁ : 1 < o) (ho : principal (+) o) :
+  o.is_limit :=
+begin
+  refine ⟨λ ho₀, _, λ a hao, _⟩,
+  { rw ho₀ at ho₁,
+    exact not_lt_of_gt ordinal.zero_lt_one ho₁ },
+  { cases eq_or_ne a 0 with ha ha,
+    { rw [ha, succ_zero],
+      exact ho₁ },
+    { refine lt_of_le_of_lt _ (ho hao hao),
+      rwa [succ_eq_add_one, add_le_add_iff_left, one_le_iff_ne_zero] } }
+end
+
+theorem principal_add_iff_add_left_eq_self {o : ordinal} :
+  principal (+) o ↔ ∀ a < o, a + o = o :=
+begin
+  refine ⟨λ ho a hao, _, λ h a b hao hbo, _⟩,
+  { cases lt_or_le 1 o with ho₁ ho₁,
+    { exact op_eq_self_of_principal hao (add_is_normal a) ho (principal_add_is_limit ho₁ ho) },
+    { rcases le_one_iff.1 ho₁ with rfl | rfl,
+      { exact (ordinal.not_lt_zero a hao).elim },
+      { rw lt_one_iff_zero at hao,
+        rw [hao, zero_add] }}},
+  { rw ←h a hao,
+    exact (add_is_normal a).strict_mono hbo }
+end
+
+theorem add_omega {a : ordinal} (h : a < omega) : a + omega = omega :=
```

Comment:
I think I'd expect to see this result in `ordinal_arithmetic`, as well as some of the ones that were there and have now been moved. Perhaps instead you can add something to the docstring of that file indicating to look here for some lemmas? (Or just keep these where they were, up to you!)

Reply:
I've added a remark at the top of the file..